rust/lance-core/src/cache.rs

```diff
@@ -107,7 +123,11 @@ impl LanceCache {
     pub fn with_key_prefix(&self, prefix: &str) -> Self {
         Self {
             cache: self.cache.clone(),
-            prefix: format!("{}{}/", self.prefix, prefix),
+            prefix: if self.prefix.is_empty() {
+                prefix.to_string()
+            } else {
+                format!("{}/{}", self.prefix, prefix)
+            },
```

Comment:
need to double check this. I didn't understand how with_key_prefix and get_key were consistent previously